pkg/interface/publish/src/js/components/lib/notebook.js

```diff
@@ -41,11 +44,31 @@ export class Notebook extends Component {
   }
 
   componentDidMount() {
+    this.canWriteCheck();
     if (this.props.notebooks[this.props.ship][this.props.book].notes) {
       this.onScroll();
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
```

Comment:
Isn't this condition a problem for deeply nested objects?

Reply:
We're not checking the whole object; just checking if any of its first levels get initialised.

Reply:
You asked the right question, and you had the right response! :) I love this